src/cpp/core/r_util/RActiveSessionStorage.cpp

```diff
@@ -26,153 +25,165 @@ namespace r_util {
-            std::string value = "";
-
-            if (readPath.exists())
-            {
-                Error error = core::readStringFromFile(readPath, &value);
-                if (error)
-                    failedFiles.push_back(readPath);
-                boost::algorithm::trim(value);
-            }
-            pValues->insert(std::pair<std::string, std::string>{name, value});
-        }
-
-        if(failedFiles.empty())
-            return Success();
-        else
-            return createError("UnableToReadFiles", "Failed to read from the following files ", 
-                failedFiles, ERROR_LOCATION);
-    }
-
-    Error FileActiveSessionStorage::readProperties(std::map<std::string, std::string>* pValues)
-    {
-        FilePath propertyDir = getPropertyDir();
-        std::vector<FilePath> files{};
-        std::vector<FilePath> failedFiles{};
-        pValues->empty();
-        propertyDir.getChildren(files);
-
-        for(FilePath file : files) {
-            std::string value = "";
-            Error error = core::readStringFromFile(file, &value);
-
-            if(error)
-                failedFiles.push_back(file);
-
-            std::string propertyName = getFileNameProperty(file.getFilename());
-            pValues->insert(std::pair<std::string, std::string>{propertyName, value});
-        }
-
-        if(failedFiles.empty())
-            return Success();
-        else
-            return createError("UnableToReadFiles", "Failed to read from the following files ",
-                failedFiles, ERROR_LOCATION);
-    }
-
-    Error FileActiveSessionStorage::writeProperty(const std::string& name, const std::string& value)
-    {
-        std::map<std::string, std::string> property = {{name, value}};
-        return writeProperties(property);
-    }
-
-    Error FileActiveSessionStorage::writeProperties(const std::map<std::string, std::string>& properties)
-    {
-        std::vector<FilePath> failedFiles{};
-        for (auto&& prop : properties)
-        {
-            FilePath writePath = getPropertyFile(prop.first);
-            Error error = core::writeStringToFile(writePath, prop.second);
+   FileActiveSessionStorage::FileActiveSessionStorage(const FilePath& scratchPath) :
+      scratchPath_ (scratchPath)
+   {
+      Error error = scratchPath_.ensureDirectory();
+      if(error)
+         LOG_ERROR(error);
+   }
+
+   const std::map<std::string, std::string> FileActiveSessionStorage::fileNames =
+      {
+         { "last_used" , "last-used" },
+         { "r_version" , "r-version" },
+         { "r_version_label" , "r-version-label" },
+         { "r_version_home" , "r-version-home" },
+         { "working_directory" , "working-dir" },
+         { "launch_parameters" , "launch-parameters" }
+       };
+
+   Error FileActiveSessionStorage::readProperty(const std::string& name, std::string* pValue)
+   {
+      std::map<std::string, std::string> propertyValue{};
```

Comment:
No need to supply an empty initializer list (here and below)
```suggestion
      std::map<std::string, std::string> propertyValue;
```